kernel/kernel-impl/src/main/java/org/sakaiproject/component/impl/BasicConfigurationService.java

```diff
@@ -625,6 +625,18 @@ protected int dereferenceConfig() {
         return rv;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public long getLong(String name, long dflt)
+    {
+        String value = getString(name);
+
+        if (StringUtils.isEmpty(value)) return dflt;
+
+        return Long.parseLong(value);
```

Comment:
This method can just be "return NumberUtils.toLong(getString(name), dflt);". It will handle NumberFormatException also.

Reply:
Thanks, I was just re-using the existing pattern and didn't put much thought into improving this. NumberUtils here would probably reduce more code.